manager/integration/tests/test_snapshot.py

```diff
@@ -0,0 +1,491 @@
+import os
+import subprocess
+import time
+import datetime
+import json
+import pandas as pd
+
+from common import client, volume_name  # NOQA
+from common import update_setting
+from common import get_self_host_id
+from common import create_and_check_volume, wait_for_volume_healthy
+from common import wait_for_rebuild_start, wait_for_rebuild_complete
+from common import create_snapshot
+from common import settings_reset # NOQA
+
+from test_basic import get_volume_endpoint
+
+from test_scheduling import get_host_replica, wait_new_replica_ready  # NOQA
+
+from common import Gi, Mi
+from common import RETRY_COUNTS, RETRY_INTERVAL
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY_IMMEDIATE_CHECK_AFTER_SNAPSHOT_CREATION  # NOQA
+from common import SETTING_SNAPSHOT_DATA_INTEGRITY_CRONJOB
+from common import SETTING_SNAPSHOT_FAST_REPLICA_REBUILD_ENABLED
+from common import SNAPSHOT_DATA_INTEGRITY_IGNORED
+from common import SNAPSHOT_DATA_INTEGRITY_ENABLED
+from common import SNAPSHOT_DATA_INTEGRITY_FAST_CHECK
+from common import SNAPSHOT_DATA_INTEGRITY_DISABLED
+
+RETRY_WAIT_CHECKSUM_COUNTS = 600
+SNAPSHOT_CHECK_PERIOD = 300
+
+def test_snapshot_hash_global_enabled_with_immediate_hash(client, volume_name, settings_reset):  # NOQA
```

Comment:
Suggest adding some descriptions or context to the test cases.

Reply:
Updated. Thank you.